modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java

```diff
@@ -88,16 +88,18 @@ public IngestDocument execute(IngestDocument document) {
                 throw new IllegalArgumentException("cannot remove metadata field [" + path + "]");
             }
             // removing _id is disallowed when there's an external version specified in the request
-            String versionType = document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class);
-            if (path.equals(IngestDocument.Metadata.ID.getFieldName())
-                && !Objects.equals(versionType, VersionType.toString(VersionType.INTERNAL))) {
-                Long version = document.getFieldValue(IngestDocument.Metadata.VERSION.getFieldName(), Long.class);
-                throw new IllegalArgumentException(
-                    "cannot remove metadata field [_id] when specifying external version for the document, version: "
-                        + version
-                        + ", version_type: "
-                        + versionType
-                );
+            if (path.equals(IngestDocument.Metadata.ID.getFieldName())) {
+                String versionType = document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class, true);
+                if (Objects.equals(versionType, VersionType.toString(VersionType.EXTERNAL))
```

Comment:
The previously implemented check was more concise and compact:
 
```suggestion
                if (!Objects.equals(versionType, VersionType.toString(VersionType.INTERNAL))
```

Reply:
We have to check if `versionType` is null, if it's null, no exception will be thrown, something like this:
```
if (versionType!=null && !Objects.equals(versionType, VersionType.toString(VersionType.INTERNAL)){
...
}
```
, do you still think this is clearer than the positive logic?

Reply:
Ah I see you changed `document.getFieldValue` to use "ignore missing" favor, may be we could do:

```
if (path.equals(IngestDocument.Metadata.ID.getFieldName()) && document.hasField(IngestDocument.Metadata.VERSION_TYPE.getFieldName()) {
```

Reply:
Thanks, that makes sense, I've changed the code.